Content/Items/Misc/Weapons.Soilgun.cs

```diff
@@ -0,0 +1,1488 @@
+                    case ItemID.PearlsandBlock: infoTooltip2 = new TooltipLine(Mod, "AmmoInfoTooltip", "When used with the Soilgun, it will fire out blocks of Pearlsand that home in on enemies"); break;
+                    case ItemID.DirtBlock: infoTooltip2 = new TooltipLine(Mod, "AmmoInfoTooltip", "When used with the Soilgun, it will fire out blocks of dirt"); break;
+                    case ItemID.SiltBlock: infoTooltip2 = new TooltipLine(Mod, "AmmoInfoTooltip", "When used with the Soilgun, it will fire out blocks of silt, that spawn coins upon hitting enemies"); break;
+                    case ItemID.SlushBlock: infoTooltip2 = new TooltipLine(Mod, "AmmoInfoTooltip", "When used with the Soilgun, it will fire out blocks of slush that cause hit enemies to have icicles impale them\nHitting and enemy with more than 15 icicles causes all icicles to shatter, causing large amounts of damage"); break;
+                }
+                tooltips.Add(infoTooltip2);
+                infoTooltip2.OverrideColor = new Color(114, 81, 56);
+            }
+        }
+    }
+
+    class SoilgunGlobalNPC : GlobalNPC
+    {
+        //its actually ice but whateva
+        public const int MaxHauntedStacks = 20;
+
+        public int GlassAMT;
+
+        public int GlassPlayerID;
+
+        public int HauntedStacks;
+
+        public int HauntedTimer;
+
+        public int HauntedSoulDamage;
+
+        public int HauntedSoulOwner;
+
+        public int SpawnHauntedSoulTimer = 60;
+        public override bool InstancePerEntity => true;
+
+        public override void ResetEffects(NPC npc)
+        {
+            if (HauntedTimer > 0)
+            {
+                HauntedTimer--;
+                if (HauntedTimer <= 0)
+                {
+                    HauntedTimer = 0;
+                    HauntedStacks = 0;
+                }
+            }
+            HauntedStacks = Utils.Clamp(HauntedStacks, 0, MaxHauntedStacks);
+        }
+        public override void AI(NPC npc)
+        {
+            if (GlassAMT > 0)
+            {
+                if (Main.rand.NextBool(20 - GlassAMT))
+                    Dust.NewDustDirect(npc.position, npc.width, npc.height, DustID.Ice, 0f, 0f, 25, default, Main.rand.NextFloat(0.8f, 1.1f));
+            }
+            if (HauntedTimer > 0)
+            {
+                float Rand = MathHelper.Clamp(20 - HauntedStacks, 1f, 20f);
+                if (Main.rand.NextBool((int)Rand))
+                    Dust.NewDustDirect(npc.position, npc.width, npc.height, DustID.Shadowflame, 0f, 0f, 25, default, Main.rand.NextFloat(0.75f, 1.2f));
+
+                if (HauntedStacks == MaxHauntedStacks)
+                {
+                    if (SpawnHauntedSoulTimer == 60)
+                        SoundEngine.PlaySound(new SoundStyle($"{nameof(StarlightRiver)}/Sounds/ShadowSpawn").WithPitchOffset(Main.rand.NextFloat(-0.1f, 0.1f)));
```

Comment:
Use Helpers.Helper.PlayPitched instead of SoundEngine.PlaySound for custom sounds